src/main/java/mate/academy/spring/service/impl/RoleServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package mate.academy.spring.service.impl;
+
+import java.util.NoSuchElementException;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.model.Role;
+import mate.academy.spring.service.RoleService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class RoleServiceImpl implements RoleService {
+    private final RoleDao roleDao;
+
+    public RoleServiceImpl(RoleDao roleDao) {
+        this.roleDao = roleDao;
+    }
+
+    @Override
+    public Role add(Role role) {
+        return roleDao.add(role);
+    }
+
+    @Override
+    public Role getByName(Role.RoleName roleName) {
```

Comment:
Sorry that was a mistake in previous review, the service method should get `String` as parameter, and the dao method should get `Role.RoleName` as parameter according the task description. Let's revert fix in this line:
```suggestion
    public Role getByName(String roleName) {
```